[docs] Update EAS Update rollouts doc (#21872)
expo/expo: docs/pages/eas-update/rollouts.mdx

hideTOC: true
---

> Rollouts allow us to deploy a new branch to a specific channel incrementally. With rollouts, we can roll out Branch B over Branch A by percentage. This is useful when we want to test a new feature to minimize the risk of introducing bugs or other issues to our production environment.

Reviewer: ```suggestion
import { Terminal } from '~/ui/components/Snippet';

Rollouts allow deploying a new branch to a specific channel incrementally. With rollouts, you can roll out Branch B over Branch A by percentage. This is useful when testing a new feature to minimize the risk of introducing bugs or other issues to your production environment.
```

- [Since this is not a tutorial, as per our writing styleguide, let's use second person voice for the whole document.](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#write-in-the-second-person)
- I'm not sure I understand the context "Branch B over Branch A by percentage." Confusing part is "by percentage". This also doesn't general tell general audience about difference between two branches. I think you might want to say, "...one branch over the other" or something like that.
- Please import the `<Terminal>` component as it'll be used in the content below for bash scripts.

> Rollouts allow us to deploy a new branch to a specific channel incrementally. With rollouts, we can roll out Branch B over Branch A by percentage. This is useful when we want to test a new feature to minimize the risk of introducing bugs or other issues to our production environment.

Reviewer 2: ```suggestion
Rollouts allow us to deploy a branch to a specific channel incrementally. With rollouts, we can roll out one branch to a percentage of end users and leave the remaining percentage of users on the current branch. This is useful when we want to test a new feature to minimize the risk of introducing bugs or other issues to our production environment.
```

> To initiate a rollout, we can use the following command:

Reviewer: ```suggestion
To initiate a rollout, use the following command:
```

Let's be instructive and eliminate "can".

> ```bash
> eas channel:rollout [channel]
> ```

Reviewer: ```suggestion
<Terminal cmd={["$ eas channel:rollout [channel]"]} />
```

Let's use `<Terminal>` component here for shell scripts: https://github.com/expo/expo/tree/main/docs#use-terminal-component-for-shell-commands-snippets

Reviewer 2: I removed the necessity for the [channel] arg (it prompts for it if not supplied)

> The interactive prompts will guide us through choosing a channel, a branch to begin rolling out, and the percentage of our users to rollout to. To increase or decrease the rollout amount, we can run the `eas channel:rollout [channel]` command again to adjust the rollout percentage.

Reviewer: ```suggestion
The interactive prompts will guide you through choosing a channel, a branch to begin rolling out, and the percentage of our users to rollout to. To increase or decrease the rollout amount, run the `eas channel:rollout [channel]` command again to adjust the rollout percentage.
```

> Once we feel confident in our new branch, or if we want to revert to our original branch, we can end the rollout with the `--end` flag:

Reviewer: ```suggestion
After you feel confident with your new branch, or if you want to revert to the original branch, you can end the rollout with the `--end` flag:
```

> ```bash
> eas channel:rollout [channel] --end
> ```

Reviewer: ```suggestion
<Terminal cmd={["$ eas channel:rollout [channel] --end"]} />
```

> Here are a variety of facts that will help us when working with rollouts:
> - We can only have one branch rolling out on a channel at once.
> - We can see the state of the rollout with `eas channel:view [channel]`.
> - When a rollout is in progress, we can still publish updates to both branches, by running `eas update --branch [branch]`, for example.
> - We cannot use `eas update --channel [channel]` when a rollout is in progress, since we cannot know which branch to associate the update with.

Reviewer: ```suggestion
Facts that will help you when working with rollouts:
- At once, you can only have one branch rolling out on a channel.
- To see the state of the rollout, use the `eas channel:view [channel]` command.
- When a rollout is in progress, you can publish updates to both branches by running `eas update --branch [branch]`.
- You cannot use `eas update --channel [channel]` when a rollout is in progress since it cannot know which branch to associate the update with.
```
